src/components/ContributorAllocations.js

```diff
@@ -68,6 +75,62 @@ const ContributorAllocations = (props) => {
     const { getContributorById: contributorAllocations } = dataContributorAllocations
     const { getContributorById: contributor } = dataContributor
 
+    const allocatedAllocations = filter(contributorAllocations.allocations, 'payment')
+    const proposedAllocations = filter(contributorAllocations.allocations, { 'payment': null })
+
+    const handleSortButton = (event) => {
+        setanchorEl(event.currentTarget)
+    }
+
+    const handleClose = () => {
+        setanchorEl(null);
+    }
+
+    const sortByNewestStartDate = () => {
+        setSortAllocated(orderBy(allocatedAllocations, ['start_date'], ['desc']))
+        setSortProposed(orderBy(proposedAllocations, ['start_date'], ['desc']))
+        setSelectedSort('Start Date (Newest first)')
+        handleClose()
+    }
+
+    const sortByOldestStartDate = () => {
+        setSortAllocated(orderBy(allocatedAllocations, ['start_date'], ['asc']))
+        setSortProposed(orderBy(proposedAllocations, ['start_date'], ['asc']))
+        setSelectedSort('Start Date (Oldest first)')
+
+        handleClose()
+    }
+
+    const sortByProjectName = () => {
+        setSortAllocated(orderBy(allocatedAllocations, item => item.project.name.toLowerCase(), ['asc']))
+        setSortProposed(orderBy(proposedAllocations, item => item.project.name.toLowerCase(), ['asc']))
+        setSelectedSort('Project Name')
+        handleClose()
+
+    }
+
+    const sortByClientName = () => {
+        setSortAllocated(orderBy(allocatedAllocations, item => item.project.client.name.toLowerCase(), ['asc']))
+        setSortProposed(orderBy(proposedAllocations, item => item.project.client.name.toLowerCase(), ['asc']))
+        setSelectedSort('Client Name')
+        handleClose()
+
+    }
+
+    const sortByPayment = () => {
+        setSortAllocated(orderBy(allocatedAllocations, ['amount'], ['desc']))
+        setSortProposed(orderBy(proposedAllocations, ['amount'], ['desc']))
+        setSelectedSort('Payment')
+        handleClose()
+    }
+
+    const clearSort = () => {
+        setSortAllocated(allocatedAllocations)
+        setSortProposed(proposedAllocations)
+        setSelectedSort(null)
+        handleClose()
+    }
+
```

Comment:
Remember DRY principle and try to make something more modular, I made this like an example but maybe there's an even better solution
```
const sortAllocations = ({typeName, allocated, proposed}) => {
        setSortAllocated(orderBy(allocated.allocations, allocated.sortingType, allocated.sortingOrder))
        setSortProposed(orderBy(proposed.alloations, proposed.sortingType, proposed.sortingOrder))
        setSelectedSort(typeName)
}
///
sortAllocations({
   typeName: 'Payment',
   allocated: {
      allocations: allocatedAllocations,
      sortingType: ['amount'],
      sortingOrder: ['desc']
   },
   proposed: { ... }
})
```

Reply:
I like this solution but it won't work for project name and client name cases.

```
const sortByProjectName = () => {
    setSortAllocated(orderBy(allocatedAllocations, item => item.project.name.toLowerCase(), ['asc']))
    setSortProposed(orderBy(proposedAllocations, item => item.project.name.toLowerCase(), ['asc']))
    setSelectedSort('Project Name')
    handleClose()
}
```

```
const sortByClientName = () => {
    setSortAllocated(orderBy(allocatedAllocations, item => item.project.client.name.toLowerCase(), ['asc']))
    setSortProposed(orderBy(proposedAllocations, item => item.project.client.name.toLowerCase(), ['asc']))
    setSelectedSort('Client Name')
    handleClose()
}
```

Reply:
@Ovifer13 I don't see those changes, please push the remaining code into GitHub